src/lintsampler/lintsampler.py

```diff
@@ -0,0 +1,547 @@
+    samples within the grid.
+    """
+
+    def __init__(self, pdf, cells, rnginput=None, vectorizedpdf=False):
+        """Initialise a LintSampler instance.
+
+        Parameters
+        ----------
+        pdf : function
+            Probability density function from which to draw samples.
+
+        cells : single array, tuple of arrays or list of tuples of arrays
+            If a single array, the boundary values for a 1D grid. If a tuple of arrays, the
+            boundary values for a grid with dimensionality of the length of the tuple. If
+            a list of tuples of arrays, the boundary values for an arbitrary number of
+            grids with dimensionality the length of the tuples.
+
+        rnginput : {None, int, ``numpy.random.Generator``}, optional
+            Seed for ``numpy`` random generator. Can be random generator itself, in
+            which case it is left unchanged. Can also be the seed to a default generator.
+            Default is None, in which case new
+            default generator is created. See ``numpy`` random generator docs for
+            more information.
+
+        vectorizedpdf : boolean
+            if True, assumes that the pdf passed is vectorized (i.e. can accept [...,k]-shaped arguments).
+            if False, assumes that the pdf passed accepts k arguments.
+
+        Returns
+        -------
+        None
+
+
+        Attributes
+        --------
+        pdf : function
+        rng : ``numpy`` random generator
+        vectorizedpdf : boolean
+
+        """
+
+        # set the pdf to be widely accessible
+        self.pdf = pdf
+
+        # set the random number generator
+        if isinstance(rnginput,np.random._generator.Generator):
+            self.rng = rnginput
+        elif rnginput==None:
+            self.rng = np.random.default_rng(42)
+        else:
+            self.rng = np.random.default_rng(rnginput)
+
+        # set the pdf optimisation flag
+        self.vectorizedpdf = vectorizedpdf
+
+        # set up the sampling grid under the hood
+        self._setgrid(cells)
+
+    def _evaluate_gridded_pdf(self,funcargs=()):
+        """Evaluate the pdf, handling the flag for vectorized.
+        
+        Parameters
+        ------------
+        self : LintSampler
+            The LintSampler instance.
+
+            
+        """
+        # tally the number of gridpoints
+        ngridpoints = np.prod(self.edgedims)
+
+        # create the flattened grid for evaluation in k>1 case
+        if self.dim > 1: 
+            edgegrid = np.stack(np.meshgrid(*self.edgearrays, indexing='ij'), axis=-1).reshape(np.prod(self.edgedims), self.dim)
+
+        # reshape the grid: assumes function takes same number of arguments as dimensions
+        if self.vectorizedpdf:
+            if self.dim > 1:
+                evalf = self.pdf(edgegrid,*funcargs).reshape(*self.edgedims)
+            else:
+                evalf = self.pdf(self.edgearrays,*funcargs)
```

Comment:
Two things here: one is that the 1D case is not yet covered in the non-vectorized route (because edgegrid is not defined in line 148), and the other is that if you accept my suggestion below to make `self.edgearrays` always a list then the 1D case needs to look a bit different. This I think is relatively succinct:
```suggestion
        # create the flattened grid for evaluation in k>1 case
        if self.dim > 1: 
            edgegrid = np.stack(np.meshgrid(*self.edgearrays, indexing='ij'), axis=-1).reshape(np.prod(self.edgedims), self.dim)
        else:
            edgegrid = self.edgearrays[0]

        # reshape the grid: assumes function takes same number of arguments as dimensions
        if self.vectorizedpdf:
            evalf = self.pdf(edgegrid,*funcargs).reshape(*self.edgedims)
```